File: app/src/main/java/com/joyhonest/sports_camera/PrivacyPolicyActivity.java
package com.joyhonest.sports_camera;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.webkit.ValueCallback;
import android.webkit.WebResourceError;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.RelativeLayout;
import android.widget.TextView;

import android.net.MailTo;
public class PrivacyPolicyActivity extends Activity {
    private ImageView backPrivacy;
    private TextView btnReload;
    private RelativeLayout errorLayout;
    private ProgressBar loadProgressBar;
    private boolean loadSuccess = true;
    private WebView webView;

    @Override
    protected void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        setContentView(R.layout.activity_privacy_policy);
        initView();
    }

    private void initView() {
        this.webView = (WebView) findViewById(R.id.webView);
        this.loadProgressBar = (ProgressBar) findViewById(R.id.loadProgressBar);
        this.errorLayout = (RelativeLayout) findViewById(R.id.error_layout);
        this.btnReload = (TextView) findViewById(R.id.btn_reload);
        this.backPrivacy = (ImageView) findViewById(R.id.backPrivacy);
        this.webView.getSettings().setJavaScriptEnabled(true);
        loadWebpage();
        this.webView.setWebViewClient(new WebViewClient() {
            @Override
            public void onReceivedError(WebView webView, int i, String str, String str2) {
                super.onReceivedError(webView, i, str, str2);
                if (Build.VERSION.SDK_INT >= 23) {
                    return;
                }
                loadSuccess = false;
                handleLoadSuccess(false);
            }

            @Override
            public void onReceivedError(WebView webView, WebResourceRequest webResourceRequest, WebResourceError webResourceError) {
                super.onReceivedError(webView, webResourceRequest, webResourceError);
                if (webResourceRequest.isForMainFrame()) {
                    loadSuccess = false;
                    handleLoadSuccess(false);
                }
            }

            @Override
            public void onPageFinished(WebView webView, String str) {
                PrivacyPolicyActivity privacyPolicyActivity = PrivacyPolicyActivity.this;
                privacyPolicyActivity.handleLoadSuccess(privacyPolicyActivity.loadSuccess);
            }

            @Override
            public boolean shouldOverrideUrlLoading(WebView webView, String str) {
                if (str.startsWith(MailTo.MAILTO_SCHEME)) {
                    sendMail(str);
                    return true;
                }
                return super.shouldOverrideUrlLoading(webView, str);
            }
        });
        this.btnReload.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                loadWebpage();
            }
        });
        this.backPrivacy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });
    }

    public void sendMail(String str) {
        Intent intent = new Intent("android.intent.action.SENDTO");
        intent.setData(Uri.parse(str));
        intent.putExtra("android.intent.extra.EMAIL", new String[]{str.substring(7)});
        startActivity(Intent.createChooser(intent, "Send Email"));
    }

    public void loadWebpage() {
        this.loadSuccess = true;
        this.webView.loadUrl("http://www.joyhonest.com/Privacy_policy_Sports_Camera.html");
        this.loadProgressBar.setVisibility(View.VISIBLE);
        this.webView.setVisibility(View.GONE);
        this.errorLayout.setVisibility(View.GONE);
    }



    public void handleLoadSuccess(boolean z) {
        this.loadProgressBar.setVisibility(View.GONE);
        if (z) {
            this.webView.setVisibility(View.VISIBLE);
            this.errorLayout.setVisibility(View.GONE);
        } else {
            this.webView.setVisibility(View.GONE);
            this.errorLayout.setVisibility(View.VISIBLE);
        }
        if (z) {
            if ((getResources().getConfiguration().uiMode & 48) == 32) {
                this.webView.evaluateJavascript("javascript:setMode('dark')", new ValueCallback<String>() {
                    @Override
                    public void onReceiveValue(String str) {
                    }
                });
            } else {
                this.webView.evaluateJavascript("javascript:setMode('light')", new ValueCallback<String>() {
                    @Override
                    public void onReceiveValue(String str) {
                    }
                });
            }
        }
    }
}

File: app/src/main/java/com/joyhonest/sports_camera/PrivacyPolicyActivity.kt
package com.joyhonest.sports_camera

import android.app.Activity
import android.content.Intent
import android.net.MailTo
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.view.View
import android.webkit.WebResourceError
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.RelativeLayout
import android.widget.TextView

class PrivacyPolicyActivity : Activity() {
    private var backPrivacy: ImageView? = null
    private var btnReload: TextView? = null
    private var errorLayout: RelativeLayout? = null
    private var loadProgressBar: ProgressBar? = null
    private var loadSuccess = true
    private var webView: WebView? = null
    override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)
        setContentView(R.layout.activity_privacy_policy)
        initView()
    }

    private fun initView() {
        webView = findViewById<View>(R.id.webView) as WebView
        loadProgressBar = findViewById<View>(R.id.loadProgressBar) as ProgressBar
        errorLayout = findViewById<View>(R.id.error_layout) as RelativeLayout
        btnReload = findViewById<View>(R.id.btn_reload) as TextView
        backPrivacy = findViewById<View>(R.id.backPrivacy) as ImageView
        webView!!.settings.javaScriptEnabled = true
        loadWebpage()
        webView!!.webViewClient = object : WebViewClient() {
            override fun onReceivedError(webView: WebView, i: Int, str: String, str2: String) {
                super.onReceivedError(webView, i, str, str2)
                if (Build.VERSION.SDK_INT >= 23) {
                    return
                }
                loadSuccess = false
                handleLoadSuccess(false)
            }

            override fun onReceivedError(webView: WebView, webResourceRequest: WebResourceRequest, webResourceError: WebResourceError) {
                super.onReceivedError(webView, webResourceRequest, webResourceError)
                if (webResourceRequest.isForMainFrame) {
                    loadSuccess = false
                    handleLoadSuccess(false)
                }
            }

            override fun onPageFinished(webView: WebView, str: String) {
                val privacyPolicyActivity: PrivacyPolicyActivity = this@PrivacyPolicyActivity
                privacyPolicyActivity.handleLoadSuccess(privacyPolicyActivity.loadSuccess)
            }

            override fun shouldOverrideUrlLoading(webView: WebView, str: String): Boolean {
                if (str.startsWith(MailTo.MAILTO_SCHEME)) {
                    sendMail(str)
                    return true
                }
                return super.shouldOverrideUrlLoading(webView, str)
            }
        }
        btnReload!!.setOnClickListener { loadWebpage() }
        backPrivacy!!.setOnClickListener { onBackPressed() }
    }

    fun sendMail(str: String) {
        val intent = Intent("android.intent.action.SENDTO")
        intent.data = Uri.parse(str)
        intent.putExtra("android.intent.extra.EMAIL", arrayOf(str.substring(7)))
        startActivity(Intent.createChooser(intent, "Send Email"))
    }

    fun loadWebpage() {
        loadSuccess = true
        webView!!.loadUrl("http://www.joyhonest.com/Privacy_policy_Sports_Camera.html")
        loadProgressBar!!.visibility = View.VISIBLE
        webView!!.visibility = View.GONE
        errorLayout!!.visibility = View.GONE
    }

    fun handleLoadSuccess(z: Boolean) {
        loadProgressBar!!.visibility = View.GONE
        if (z) {
            webView!!.visibility = View.VISIBLE
            errorLayout!!.visibility = View.GONE
        } else {
            webView!!.visibility = View.GONE
            errorLayout!!.visibility = View.VISIBLE
        }
        if (z) {
            if (resources.configuration.uiMode and 48 == 32) {
                webView!!.evaluateJavascript("javascript:setMode('dark')") { }
            } else {
                webView!!.evaluateJavascript("javascript:setMode('light')") { }
            }
        }
    }
}